lib/elixir/lib/calendar/iso.ex

```diff
@@ -447,4 +463,139 @@ defmodule Calendar.ISO do
       {days + days_offset, {parts, ppd}}
     end
   end
+
+  # :calendar.date_to_gregorian_days(year, month, day) - 365
+  defp date_to_gregorian_days(year, month, day) do
+    last_day = days_in_month(year, month)
+    if day <= last_day do # TODO Why does :calendar have this check?
+      days_in_prev_years(year) + days_before_month(month) + leap_day_offset(year, month) + day - 1
+    end
```

Comment:
The erlang version raises though if it doesn't match. Our returns nil. I think the erlang exists as an assertion, it should never be violated. It would be equivalent to writing:

    true = day <= last_day